implementation

berkebakkalogluosf opened the pull request.

berkebakkalogluosf committed:
implementation

berkebakkalogluosf committed:
moved trigger logic to a handler class

berkebakkalogluosf committed:
major overhaul

actually works this time

romanbychkoosf commented on ContactTriggerHandler.cls:
```diff
@@ -0,0 +1,87 @@
+public with sharing class ContactTriggerHandler {
+    public static void handleBefore(List<Contact> newRecords, Map<Id, Contact> oldRecords) {
+        
+        List<AggregateResult> primaryContactPerAccount = [SELECT AccountId From Contact WHERE Is_Primary_Contact__c = TRUE GROUP BY AccountId];
```
please add try/catch here

romanbychkoosf commented on ContactTriggerHandler.cls:
```diff
@@ -0,0 +1,87 @@
+                if(newRecord.Is_Primary_Contact__c && newRecord.AccountId != NULL) {
+                    // Check if Account already has a Primary Contact
+                    if(accountsWithPrimaryContacts.contains(newRecord.AccountId)) {
+                        newRecord.Is_Primary_Contact__c.addError('There already exists a primary contact for this account');
```
please use custom label or constant for it

romanbychkoosf commented on ContactTriggerHandler.cls:
```diff
@@ -0,0 +1,87 @@
+                    }
+                }
+            } else {
+                Contact oldRecord = oldRecords.get(newRecord.Id);
```
Please create some smaller methods for those ifs

romanbychkoosf commented on ContactTriggerHandler.cls:
```diff
@@ -0,0 +1,87 @@
+                        // Check if the Account already has a Primary Contact
+                        if(accountsWithPrimaryContacts.contains(newRecord.AccountId)) {
+                            Contact existingPrimaryContact = [SELECT Id FROM Contact WHERE AccountId = :newRecord.AccountId AND Is_Primary_Contact__c = TRUE];
+                            System.debug('existingPrimaryContact.Id' + existingPrimaryContact.Id);
```
please remove unneeded debug logs

romanbychkoosf commented on ContactTriggerHandler.cls:
```diff
@@ -0,0 +1,87 @@
+    
+    public static void multipleAccountPrimaryCheck(Map<Id, String> accountsPrimaryPhone, Contact newRecord) {
+        if(accountsPrimaryPhone.containsKey(newRecord.AccountId)) {
+            newRecord.Is_Primary_Contact__c.addError('Cannot insert / update more than one Primary Contact per Account');
```
please use constant or custom label for all texts

berkebakkalogluosf committed:
made the requested changes

- added try/catch to query
- used constants for the error strings
- refactored the if statements for before update
- removed the debug lines
